include/clad/Differentiator/DerivativeBuilder.h

```diff
@@ -411,6 +414,25 @@ namespace clad {
     clang::Expr* m_Result;
     /// A flag indicating if the Stmt we are currently visiting is inside loop.
     bool isInsideLoop = false;
+    /// Output variable of vector-valued function
+    std::string outputArrayStr;
+    unsigned int outputArrayCursor = 0;
+    unsigned int numParams = 0;
+    bool isVectorValued = false;
+
+    std::string funcPostfix() const {
+      if (isVectorValued)
+        return "_jac";
+      else
+        return "_grad";
+    }
+
+    std::string resultArg() const {
```

Comment:
```suggestion
    const char* resultArg() const {
```
This prevents a string copy.